bundle/src/main/java/com/adobe/acs/commons/util/OverridePathSlingRequestWrapper.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.util;
+
+import com.adobe.cq.sightly.WCMBindings;
+import com.day.text.Text;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.adapter.AdapterManager;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.apache.sling.api.wrappers.SlingHttpServletRequestWrapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Request wrapper to get the Resource and SlingBindings objects for a request scoped to a specified path.
+ *
+ * Among other things, this enables injection of sling models from a SlingHttpServletRequest
+ * object rather than simply a Resource, which is required by some sling models for full
+ * functionality.
+ */
+public class OverridePathSlingRequestWrapper extends SlingHttpServletRequestWrapper {
+    private static final String ATTR_SLING_BINDINGS = SlingBindings.class.getName();
+
+    private final SlingBindings myBindings = new SlingBindings();
+    private final Resource resource;
+
+    private final AdapterManager adapterManager;
+    private final Map<Class<?>, Object> adaptersCache = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param wrappedRequest SlingHttpServletRequest
+     * @param relPath        String
+     */
+    public OverridePathSlingRequestWrapper(final SlingHttpServletRequest wrappedRequest, final String relPath) {
+        super(wrappedRequest);
+
+        SlingBindings slingBindings = (SlingBindings) getSlingRequest().getAttribute(ATTR_SLING_BINDINGS);
+
+        this.adapterManager = slingBindings.getSling().getService(AdapterManager.class);
+        this.resource = getSlingRequest().getResourceResolver().resolve(getSlingRequest(),
```

Comment:
Could this be simplified to: 

`getSlingRequest().getResource().getChild(relPath);` ?

The use of `resolve(..)` and `Text.fullFilePath(..)` make it seem there is more complexity in resolving the relative path, but im not clear what that is (or if this isnt around about way of doing the above).

https://sling.apache.org/apidocs/sling7/org/apache/sling/api/resource/Resource.html#getChild-java.lang.String-

Reply:
I believe @adamcin did it this way to ensure null safety of the retrieved resource in the case that a path does not exist, since `resolve` will return a synthetic resource in that case.

Reply:
Also, `Text.fullFilePath(..)` is used to support both relative and absolute paths.  I'll update the var name and docs accordingly.

Reply:
so returning a non existing resource is a valid use case? When would requestPathInfo.getResourcePath() be relative? 

I think adding some comments to the code would be helpful clarifying these seemingly? corner cases.

Reply:
Updated OverridePathSlingRequestWrapper to no longer support relative paths, as it was an unnecessary feature and more likely to cause bugs then be truly helpful.

Added test for OverridePathSlingRequestWrapper to ensure support for requests to non-existent paths, since sling seems to generally support this concept and the only other option would be to throw an exception from the constructor.